### PR TITLE
Add aad authentication to iOS sdk

### PR DESCRIPTION
**Pull request by manuchandel**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR is for supporting AAD Enabled Authentication for iOS SDK.

### Prerequisites

1. MSAL requires **authority_url** for authentication.
2. For B2C **authority_url** is generated by Azure when a user registers his app on Azure Portal.
3. For AAD **authority_url** is different and is not returned by Azure as is.

The JSON Payload for AAD looks something like this

```
{
   "client_id" : "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX",
   "authorization_user_agent" : "DEFAULT",
   "authorities" : [
      {
         "type": "AAD",
         "audience": {
            "type": "AzureADandPersonalMicrosoftAccount",
            "tenant_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX"
         }
      }
   ]
}
```

To calculate authority_url we use following logic

1. If **type** in **audience** is **AzureADMyOrg** then uri is *https://login.microsoftonline.com/{tenantId}*
1. If **type** in **audience** is **AzureADMultipleOrgs** then uri is *https://login.microsoftonline.com/organizations*
1. If **type** in **audience** is **AzureADandPersonalMicrosoftAccount** then uri is *https://login.microsoftonline.com/common*

### MSAuthority
1. **MSAuthority** is base class of all Authorities and contains a static method **AuthorityWithDictionary** which creates instance of Authority.
1. It processes each authority and based on type initiates objects.
1. Common code processing of reading type and default is done in this class.

### MSB2CAuthority
1. **MSB2CAuthority** implements **MSAuthority** and reads **redirect_uri** as before.
1. IsValid method now has one additional check from before. it checks for the type B2C.

### MSAADAuthority
1. **MSAADAuthority** implements **MSAuthority** and constructs **authority_url** by the logic explained above.
1. IsValid method now has one additional check from before. it checks for the type AAD.

### MSAuthConfig
1. Instead of directly initiating object of MSAuthority type as before, it now calls **AuthorityWithDictionary** static method to get proper object according to authentication type.
2. Instead of specifically checking for authority type in **AreAuthoritiesValid** it now just calls **IsValid** on each authority object and that object depending on what authority it is checks for its type in runtime.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import <MSAuthority.h>
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]:
```suggestion

#import "MSAuthority.h"
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSAADAuthority : MSAuthority {
+}
+@end
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]:
```suggestion

@end
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAADAuthority.h"
+#import "../Util/MSAuthConstants.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]:
```suggestion
#import "MSAuthConstants.h"
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+@implementation MSAADAuthority
+
+static NSString *const kMSAudienceType = @"type";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]:
```suggestion
static NSString *const kMSType = @"type";
```

Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Please consider to reuse common constants.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  if (!dictionary) {
+    return nil;
+  }
+  if((self = [super initWithDictionary:dictionary])){
+    if (dictionary[kMSAudience]) {
+      if ([(NSDictionary *)dictionary[kMSAudience] isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *audience = (NSDictionary *)dictionary[kMSAudience];
+        NSString *tenantId = (NSString * _Nonnull)audience[kMSTenantId];
+        NSString *audienceType = (NSString * _Nonnull)audience[kMSAudienceType];
+        NSString *authorityUrlPath = @"common";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Consider defining a constant for the string literal.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        if ([audienceType isEqualToString:kMSSingleTenantAudience]) {
+          authorityUrlPath = tenantId;
+        } else if ([audienceType isEqualToString:kMSMultiTenantAudience]) {
+          authorityUrlPath = @"organizations";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Consider defining a constant for the string literal.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m [hunk 7]
```diff
@@ -47,10 +45,10 @@ - (BOOL)isValid {
 - (BOOL)areAuthoritiesValid {
   BOOL foundDefault = NO;
   for (MSAuthority *authority in self.authorities) {
-    if (![authority isValid]) {
+    if ([authority isValid]) {
       return NO;
     }
-    if (authority.defaultAuthority && [authority.type isEqualToString:kMSAuthorityTypeB2C]) {
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m]: Seems you need to override `areAuthoritiesValid:` in AAD and B2C Authority classes and compare `authority.type`.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 8]
```diff
@@ -2,32 +2,27 @@
 // Licensed under the MIT License.
 
 #import "MSAuthority.h"
+#import "MSB2CAuthority.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: You should sort `import`s. Please run `./clang-format-changed-files.sh` in your repo root to format all files that you touched.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 9]
```diff
@@ -2,32 +2,27 @@
...
+#import "MSAADAuthority.h"
+#import "../Util/MSAuthConstants.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]:
```suggestion
#import "MSAuthConstants.h"
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 10]
```diff
@@ -36,4 +31,18 @@ - (BOOL)isValid {
   return self.type && self.authorityUrl;
 }
 
++ (MSAuthority *)authorityWithDictionary:(NSDictionary *)dictionary {
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]:
```suggestion
+ (instancetype)authorityWithDictionary:(NSDictionary *)dictionary {
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h [hunk 11]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import <MSAuthority.h>
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]:
```suggestion

#import "MSAuthority.h"
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSB2CAuthority : MSAuthority {
+}
+@end
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]:
```suggestion

@end
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m [hunk 13]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSB2CAuthority.h"
+#import "../Util/MSAuthConstants.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]:
```suggestion
#import "MSAuthConstants.h"
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m (hunk 10 above)
Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: Do you really need this utility method? This will couple the base class with its children.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h [hunk 14]
```diff
@@ -25,3 +25,13 @@ static NSString *const kMSAuthConfigFilename = @"config.json";
  * The eTag key to store the eTag of current configuration.
  */
 static NSString *const kMSAuthETagKey = @"MSAuthETagKey";
+
+/**
+ * B2C authority type
+ */
+static NSString *const kMSAuthorityTypeB2C = @"B2C";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: Those constants are very specific to AAD and B2C authorities. I would suggest to move this to each authority class and remove the utility method in the base class. You can check `type` in a dictionary in `MSAuthConfig` at the time it instantiate an object.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m [hunk 15]
```diff
@@ -47,10 +45,10 @@ - (BOOL)isValid {
 - (BOOL)areAuthoritiesValid {
   BOOL foundDefault = NO;
   for (MSAuthority *authority in self.authorities) {
-    if (![authority isValid]) {
```
Comment (dhei) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m]: Is the `!` removed intentionally?

Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m]: No it was intentional I will fix it.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h [hunk 16]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import <MSAuthority.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSAADAuthority : MSAuthority {
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]: You don't need `{}`s for interfaces, just the `@end` marker

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+static NSString *const kMSMultiTenantAudience = @"AzureADMultipleOrgs";
+
+static NSString *const kMSCommonAudience = @"AzureADandPersonalMicrosoftAccount";
+
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+  if (!dictionary) {
+    return nil;
+  }
+  if((self = [super initWithDictionary:dictionary])){
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Remember to run the clang formatter

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+// Licensed under the MIT License.
+
+#import "MSAADAuthority.h"
+#import "../Util/MSAuthConstants.h"
+
+@implementation MSAADAuthority
+
+static NSString *const kMSAudienceType = @"type";
+
+static NSString *const kMSAuthorityUrl = @"authority_url";
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Lots of these constants are keys so they should indicate that:

```suggestion
static NSString *const kMSAuthorityUrlKey = @"authority_url";
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+
+static NSString *const kMSAudience = @"audience";
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]:
```suggestion
static NSString *const kMSAudienceKey = @"audience";
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+
+@implementation MSB2CAuthority
+
+static NSString *const kMSType = @"type";
```
Comment (dhei) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]: We can move "type" and "authority_url" constants to the parent class, right?

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+
+static NSString *const kMSTenantId = @"tenant_id";
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]:
```suggestion
static NSString *const kMSTenantIdKey = @"tenant_id";
```

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m (hunk 10 above)
Comment (jwallra) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: Shouldn't this be named `initWithDictionary` still?

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h [hunk 22]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import <MSAuthority.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSB2CAuthority : MSAuthority {
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]: Don't need `{}`s

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m (hunk 20 above)
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]: These should be reused from `MSAADAuthority` - put them in a common place

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m (hunk 4 above)
Comment (dhei) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: `kMSAudienceType` and `kMSAuthorityUrl` can be moved to parent class or common constants.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h (hunk 2 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]: formatted

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h (hunk 1 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]: changed import types

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m (hunk 3 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: no longer valid

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m (hunk 10 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: I can remove this method and add whole logic to MSAuthConfig if that seems more convenient.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h (hunk 11 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]: removed angle braces

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m (hunk 13 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]: no longer valid

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 14 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: Got it. But if check type in MSAuthConfig. I would require those constants again in this file. Can you please suggest where should I put these constants so that they are accessible to all files in a folder?

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m (hunk 15 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m]: No it is not. But thanks for catching this.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h (hunk 16 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.h]: yeah. didn't realize. :)

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m (hunk 5 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: defined constants for key literals.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h (hunk 12 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]: added a line

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m (hunk 20 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]: No its not. Removed it.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m (hunk 18 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: changed constants to keys

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m (hunk 21 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: changed constants to keys

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h (hunk 22 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.h]: yes removed.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m (hunk 20 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSB2CAuthority.m]: Can you please suggest this common place? Should I add this to MSAuthConstants?

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+  if (!dictionary) {
+    return nil;
+  }
+  if ((self = [super initWithDictionary:dictionary])) {
+    if (dictionary[kMSAudienceKey]) {
+      if ([(NSDictionary *)dictionary[kMSAudienceKey] isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *audience = (NSDictionary *)dictionary[kMSAudienceKey];
+        NSString *tenantId = (NSString * _Nonnull)audience[kMSTenantIdKey];
```
Comment (jwallra) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: Do we need to type-check the contents for these fields or do we just take whatever is in the dictionary and fail later?

Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m]: In android we are trusting the contents and passing along. I guess we can do that here. MSAL will take careof contents.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 14 above)
Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: You can define those in each class and reference it from there.

File: AppCenterAuth/AppCenterAuth.xcodeproj/xcshareddata/xcschemes/AppCenterAuth macOS Framework.xcscheme [hunk 24]
```diff
@@ -29,7 +29,26 @@
       codeCoverageEnabled = "YES"
       shouldUseLaunchSchemeArgsEnv = "YES">
       <Testables>
+         <TestableReference
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth.xcodeproj/xcshareddata/xcschemes/AppCenterAuth macOS Framework.xcscheme]: Please revert this change and merge develop when #1775 is merged.

File: AppCenterAuth/AppCenterAuth.xcodeproj/xcshareddata/xcschemes/AppCenterAuth iOS Framework.xcscheme [hunk 25]
```diff
@@ -29,7 +29,26 @@
       codeCoverageEnabled = "YES"
       shouldUseLaunchSchemeArgsEnv = "YES">
       <Testables>
+         <TestableReference
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth.xcodeproj/xcshareddata/xcschemes/AppCenterAuth iOS Framework.xcscheme]: Please revert this change and merge develop when #1775 is merged.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 26]
```diff
@@ -2,29 +2,30 @@
 // Licensed under the MIT License.
 
 #import "MSAuthority.h"
+#import "MSAADAuthority.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: Why are you importing this one here?

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 27]
```diff
@@ -2,29 +2,30 @@
...
+#import "MSAuthConstants.h"
+#import "MSB2CAuthority.h"
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: Same here.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h [hunk 28]
```diff
@@ -25,3 +25,18 @@ static NSString *const kMSAuthConfigFilename = @"config.json";
...
+
+/**
+ * The B2C authority type string.
+ */
+static NSString *const kMSAuthorityTypeB2C = @"B2C";
+
+/**
+ * The provider type key.
+ */
+static NSString *const kMSTypeKey = @"type";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: Move this to `MSAuthority.h`. This is only used by Authority classes.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h [hunk 29]
```diff
@@ -25,3 +25,18 @@ static NSString *const kMSAuthConfigFilename = @"config.json";
  * The eTag key to store the eTag of current configuration.
  */
 static NSString *const kMSAuthETagKey = @"MSAuthETagKey";
+
+/**
+ * The AAD authority type string.
+ */
+static NSString *const kMSAuthorityTypeAAD = @"AAD";
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: Move this to `MSAADAuthority.h`.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 14 above)
Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: Move this to `MSB2CAuthority.h`.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m [hunk 30]
```diff
@@ -36,4 +37,7 @@ - (BOOL)isValid {
   return self.type && self.authorityUrl;
 }
 
+- (BOOL)isValidType {
+  return NO;
+}
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthority.m]: Line break after this line.

File: AppCenterAuth/AppCenterAuthTests/MSAuthConfigTest.m [hunk 31]
```diff
@@ -79,21 +92,48 @@ - (void)testConfigIsValid {
...
   XCTAssertTrue([config isValid]);
+
+  // When
+  b2cAuth.type = @"notB2C";
+
+  // Then
+  XCTAssertFalse([config isValid]);
+
+  // When
+  MSAuthority *aadAuth = [MSAADAuthority new];
```
Comment (jaelim-ms) [inline AppCenterAuth/AppCenterAuthTests/MSAuthConfigTest.m]: This is `IsValid` test for AAD. Please split the test into two separate tests.

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 29 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: I cannot define constants in header files unless I use extern keyword. Do you want me to use extern keyword for this?

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 28 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: This is also used by MSAuthConfig to find out type and instantiate object

File: AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h (hunk 14 above)
Comment (manuchandel) [reply AppCenterAuth/AppCenterAuth/Internals/Util/MSAuthConstants.h]: This is also used by MSAuthConfig to instantiate object of type of authority.

File: AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m (hunk 7 above)
Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m]: Never mind. I saw this as in `MSAuthority`.
